src/shared/components/oncoprint/DataUtils.ts

```diff
@@ -199,18 +212,25 @@ export function makeGeneticTrackData(
             newDatum.uid = sample.uniqueSampleKey;
 
             const sampleSequencingInfo = genePanelInformation.samples[sample.uniqueSampleKey];
-            newDatum.profiled_in = sampleSequencingInfo.byGene[hugoGeneSymbol] || [];
+            newDatum.profiled_in = _.flatMap(
+                geneSymbolArray,
+                hugoGeneSymbol => sampleSequencingInfo.byGene[hugoGeneSymbol] || []
+            );
             newDatum.profiled_in = newDatum.profiled_in.concat(sampleSequencingInfo.allGenes);
             if (!newDatum.profiled_in.length) {
                 newDatum.na = true;
             }
-            newDatum.not_profiled_in = sampleSequencingInfo.notProfiledByGene[hugoGeneSymbol] || [];
+            newDatum.not_profiled_in = _.flatMap(
+                geneSymbolArray,
+                hugoGeneSymbol => sampleSequencingInfo.notProfiledByGene[hugoGeneSymbol] || []
+            );
             newDatum.not_profiled_in = newDatum.not_profiled_in.concat(sampleSequencingInfo.notProfiledAllGenes);
 
-            fillGeneticTrackDatum(
-                newDatum, hugoGeneSymbol,
+            _.assign(newDatum, fillGeneticTrackDatum(
+                {},
+                geneSymbolArray.join(' / '),
```

Comment:
I tried to keep the impact of the change small, but in effect I changed the semantics of all downstream code by assuming that the ‘hugoGeneSymbol’ field holds a label to be used verbatim in views rather than an actual HGNC-approved gene symbol to integrate with web APIs. It would make sense to either change the name of the attribute to make that explicit, or, if the code downstream of this object isn't strictly limited to view logic, adjust all of that code to expect an array. Do you think the latter choice is the right one then? Note that actual alterations in the `data` property still list the details of the genes they were measured in.

Reply:
I see your point. I think the `gene` property of `GeneticTrackDatum` is only used in the tooltip. So I think it would be a good idea to leave the code you've written but change the attribute name to `trackLabel`